mqtt/mqtt-broker/tests/smoke.rs

```diff
@@ -22,7 +27,7 @@ async fn basic_connect_clean_session() {
         .build();
 
     assert_matches!(
-        client.events_receiver.recv().await,
+        client.next().await,
```

Comment:
why did this change?

Reply:
@dmolokanov suggested instead of exposing receiver, to impl `Stream` for `TestClient`. It allows us to use some helper methods like `.take()` `.collect()` `.filter_map()` etc.